packages/flutter/lib/src/rendering/object.dart

```diff
@@ -471,7 +471,10 @@ class PaintingContext extends ClipContext {
     assert(clipBehavior != null);
     final Rect offsetBounds = bounds.shift(offset);
     final RRect offsetClipRRect = clipRRect.shift(offset);
-    if (needsCompositing) {
+    if (clipBehavior == Clip.none) {
```

Comment:
right now the context asserts if I push clip.none, but I think I should remove that and just make it do the right thing here

Reply:
Isn't the idea there to avoid compositing a layer unnecessarily?

Reply:
pushClip<X> only conditionally composites, take a look at the implementation. Its why we pass in needsCompositing

Reply:
oh wait, this is the implementation lol. NVM. I added this change because otherwise the ClipRRectLayer would assert